cloudwatch_metrics_collector/include/cloudwatch_metrics_collector/metrics_collector.hpp

```diff
@@ -45,7 +47,15 @@ class MetricsCollector : public Service
    * @param metric_list_msg
    * @return the number of metrics successfully batched
    */
-  int RecordMetrics(const ros_monitoring_msgs::msg::MetricList::UniquePtr metric_list_msg);
+  int RecordMetrics(ros_monitoring_msgs::msg::MetricList::UniquePtr metric_list_msg);
+
+  /**
+   * Accept input metric message to be batched for publishing.
+   *
+   * @param msg
+   * @return the number of metrics successfully batched
+   */
+  int RecordMetrics(metrics_statistics_msgs::msg::MetricsMessage::UniquePtr msg);
```

Comment:
Not const? Shouldn't this just create a MetricObject and not modify input data?

Reply:
Sure, I can change it to a `metrics_statistics_msgs::msg::MetricsMessage::ConstUniquePtr`. (I don't think it should be a `const metrics_statistics_msgs::msg::MetricsMessage::UniquePtr` or `const metrics_statistics_msgs::msg::MetricsMessage::ConstUniquePtr` though.)

Reply:
It turns out `create_subscription()` doesn't support this.